docs/integration-tutorial.md

```diff
@@ -0,0 +1,212 @@
+
+To follow this tutorial more interactively, please clone the following github repository containing all the relevant code being referenced.
+
+```sh
+git clone https://github.com/gnosis/dex-integration-tutorial
+cd dex-integration-tutorial
+```
+
+This guide will focus primarily on interacting with dex-contracts (i.e. the Gnosis Protocol Smart contracts) and writing a minimally functional integration with synthetix exchange. Note that the README in the cloned repository may be updated with additional integrations (such as Uniswap) in the future.
+
+## Project Initialization and Network Configuration
+
+This section is intended to be a first step when building a new project from scratch. If you would prefer to start from a preconfigured environment, feel free to skip ahead to the next section.
+
+```sh
+mkdir <project_title>
+cd project_title
+truffle init
+yarn add @truffle/contract
+yarn add @gnosis.pm/dex-contracts
+[optional] yarn add @gnosis.pm/util-contracts
+[optional] yarn add @openzeppelin/contracts@2.5.1
+```
+
+Make sure that you have a valid (truffle configuration)[https://www.trufflesuite.com/docs/truffle/reference/configuration] for the network you intend on interacting with.
+We will be using the common `truffle-config.js` found in most, if not all, gnosis smart contract projects.
+In order to use the common gnosis truffle configuration, you will need to install the optional package `@gnosis.pm/util-contracts` listed above.
+Furthermore, you will likely have to provide your own `INFURA_KEY`
+
+Note that, if you plan to be experimenting with a locally hosted development network, you will need to install additional "devDepencencies" to mirgrate the `BatchExchange` Smart Contracts. This will be covered in detail once we have successfully confirmed our ability to interact with the existing mainnet smart contracts.
+
+## Interaction with Batch Exchange
+
+Assuming you were successful with optionally configuring your own project from the section above, we will continue here from a pre-configured environment obtained and installed as follows:
+
+```sh
+git clone git@github.com:bh2smith/dex-integration-tutorial.git
+cd dex-integration-tutorial
+yarn
+```
+
+This should put us in the same place as has having completed the project initialization steps independantly.
+We are now prepared to start scripting interactions with the Gnosis Protocol.
+To test this run the `exchange_interaction` script via
+
+Note that before attempting to execute any of these truffle scripts, you will have to export your own `INFURA_KEY`!
+
+```sh
+export INFURA_KEY=<your infura key>
+```
+
+For less repitiion, this can also be included directly in the truffle configuration file at [line 17](https://github.com/bh2smith/dex-integration-tutorial/blob/44320ed3345ddb72ea5f0884460e8e1789ad48fe/truffle-config.js#L17). Just replace the existing one after the `||`. Be carefull not to commit your infura key!
```

Comment:
I would not suggest this overhere, since committing the infura key happens sooo quickly.